webbeans-web/src/main/java/org/apache/webbeans/web/context/ServletRequestContext.java

```diff
@@ -32,7 +32,7 @@
     private static final long serialVersionUID = -8375349845543590243L;
 
     // this can only be accessed when the context is active
-    private transient HttpServletRequest servletRequest;
+    private transient ServletRequest servletRequest;
```

Comment:
this change is weird to me because WebContextsService is about supporting the spec and part of it request and session  scopes. It means the init event can't be a ServletRequest but must be a HttpServletRequest or we are in one of these cases:

1. we don't run in a servlet container
2. OWB listeners are wrongly set up and don't see the right request passing (it is already wrapped)
3. OWB filter is set after a filter which wraps the request and breaks "Http" part of it

IMHO it is 2 cases we don't want to support in webbeans-web but should make simpler to support.

Maybe some more context about how tomee can get it would help to refine which case we are in and get a more accurate fix?